channels/OFF-CHAIN.md

```diff
@@ -291,23 +301,96 @@ different amounts, then that might communicate that it wants these instead.)
   blockchain to stop a cheater
 - `responder_pubkey` MUST be a valid ed25519 pubkey
 
+### `channel_reestabl`
 
-### `funding_created`
+Message code: 3
+
+It is possible to resume a channel which has been terminated either due
+to client failure or by the participants mutually agreeing to leave.
+
+```
+  name                  size (bytes)
+ ---------------------- ----
+| chain_hash           | 32 |
+ ---------------------- ----
+| channel_id           | 32 |
+ ---------------------- ----
+| length               | 16 |
+ ---------------------- ----
+| data                 | N  |
+ ---------------------- ----
+```
+
+The payload (`data`) must be the latest mutually signed offchain state, and
+the clients must verify that they have the corresponding state trees to match
+the state (otherwise, it will not be possible to close the channel later on.)
```

Comment:
I think the text in brackets could be improved a little bit: no matter the state trees, participants can always make a `channel_close_mutual` that only has the closing balances they've agreed upon (no state hash there).
If they are using different state trees (not matching the `state_hash` provided in the off-chain transaction in `data`) they'd be performing the same updates on different trees and will getting different root hashes for those trees. This would make the channel unusable.
My suggestion would be:
(otherwise, it will not be possible to ~close~ use the channel later on.)